Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/BufferDescriptor.h

```diff
@@ -22,47 +24,44 @@ namespace AZ
          * A set of combinable flags which inform the system how a buffer is to be
          * bound to the pipeline at all stages of its lifetime.
          */
-        enum class BufferBindFlags : uint32_t
-        {
-            None            = 0,
+        AZ_ENUM_CLASS_WITH_UNDERLYING_TYPE(BufferBindFlags, uint32_t,
```

Comment:
Is there any added advantage of using AZ_ENUM_CLASS_WITH_UNDERLYING_TYPE?

Reply:
Using the macro gives us a way to easily convert the bitpacked image/buffer bind flags to their string representation. Previously I was doing the naive solution of just hardcoding each enum value/string pair which is not really a maintainable pattern. With the macro we can instead get pairs of `<value, string>` by iterating over `ImageBindFlagsMembers`, which will adapt to any future changes in `BufferBindFlags` or `ImageBindFlags`.

Reply:
oh i see, nice!